examples/codeviewer/common/src/jvmMain/kotlin/org/jetbrains/codeviewer/platform/JvmFile.kt

```diff
@@ -58,22 +64,36 @@ fun java.io.File.toProjectFile(): File = object : File {
             override val size get() = size
 
             override fun get(index: Int): String {
-                val startPosition = lineStartPositions[index]
-                val length = if (index + 1 < size) lineStartPositions[index + 1] - startPosition else
-                    byteBufferSize - startPosition
-                // Only JDK since 13 has slice() method we need, so do ugly for now.
-                byteBuffer.position(startPosition)
-                val slice = byteBuffer.slice()
-                slice.limit(length)
+                val position = lineRange(index)
+                val slice = byteBuffer.slice(position.first, position.last - position.first)
                 return StandardCharsets.UTF_8.decode(slice).toString()
             }
+
+            private fun lineRange(index: Int): IntRange {
+                val startPosition = lineStartPositions[index]
+                val nextLineIndex = index + 1
+                var endPosition = if (nextLineIndex < size) {
+                    // Remove \n itself from the range
+                    lineStartPositions[nextLineIndex] - 1
+                } else byteBufferSize
```

Comment:
```suggestion
                } else byteBufferSize - 1
```

Should we also substract in this case?

Reply:
No, it's only for line separator, the character _between lines_

Reply:
Ok. What do you think if we replace it by this?

```
if (endPosition > startPosition && byteBuffer[endPosition - 1].isChar('\n')) {
    endPosition--
}
```

It is more readable

Reply:
This is also good